app/tests/workstations_tests/test_session_control.py

```diff
@@ -0,0 +1,87 @@
+import json
+from pathlib import Path
+
+import pytest
+from django.conf import settings
+from django.views.generic import TemplateView
+
+import tests.settings as test_settings
+from tests.utils import playwright_trace
+
+
+class SessionControlView(TemplateView):
+    url_route = "session-control/"
+    template_name = "session_control.html"
+
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data()
+        context.update(
+            {"domain": test_settings.DJANGO_LIVE_TEST_SERVER_ADDRESS}
+        )
+        return context
+
+
+class WorkstationView(TemplateView):
+    url_route = "workstation/"
+    template_name = "workstation.html"
+
+
+class SessionCreationView(TemplateView):
+    url_route = "new-session/"
+    template_name = "new_session.html"
+
+
+@pytest.mark.playwright
+def test_viewer_session_control(playwright_live_server, page):
+    with playwright_trace(page.context, directory=Path("/app/test_results")):
```

Comment:
You should be able to achieve the same thing with a pytest fixture rather than using a big context manager.

Reply:
That was the first version: however that would always require a context as (first) positional argument. The context manager is more dynamic in that you can, at any point, provide it with context.